edb/server/protocol/auth_helpers.pyx

```diff
@@ -235,3 +232,31 @@ cdef auth_basic(tenant, username: str, password: str):
     verifier, mock_auth = scram_get_verifier(tenant, username)
     if not scram_verify_password(password, verifier) or mock_auth:
         raise errors.AuthenticationError('authentication failed')
+
+
+cdef auth_mtls(transport):
+    sslobj = transport.get_extra_info('ssl_object')
+    if sslobj is None:
+        raise errors.AuthenticationError(
+            "mTLS authentication is not supported over plaintext transport")
+    cert_data = sslobj.getpeercert()
+    if not cert_data:  # None or empty dict
+        raise errors.AuthenticationError(
+            "valid client certificate required")
```

Comment:
Could you add a comment to the effect that peer certs are always verified against the configured client_ca_file, and so all we need to do is check that one exists?